### PR TITLE
Introduce max module dependency check

### PR DESCRIPTION
**Pull request by mkorszun**
This introduces a feature described in https://github.com/rrrene/credo/issues/621.
It is a check to warn whenever modules have more dependencies than configured (defaults to `10`).

Configuration options:

* `max_deps` - Maximum number of module dependencies

* `dependency_namespaces` - Specify which namespaces should be included while counting module dependencies - by default it will count all dependencies (project modules, built-in modules, external dependencies). If you only care about project modules, specify `[MyAppNamespace]`

* `excluded_namespaces` - Specify which namespaces within a project to exclude from this check

* `excluded_paths` - Specify paths within a project to exclude from this check - defaults to `["test"]`

### CODE REVIEW

File: lib/credo/check/refactor/module_dependencies.ex [hunk 1]
```diff
@@ -0,0 +1,143 @@
+defmodule Credo.Check.Refactor.ModuleDependencies do
+  @moduledoc false
+
+  @checkdoc """
+  Module is most probably doing too much. Consider limiting the number of
```
Comment (rrrene) [inline lib/credo/check/refactor/module_dependencies.ex]: Let's rephrase this to be less confident. Maybe:

> This module might be doing too much.

File: lib/credo/check/refactor/module_dependencies.ex [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+  defp issues_for_module(_, _, _, _, _), do: []
+
+  defp with_fullnames(dependencies, aliases) do
+    dependencies
+    |> Enum.map(&full_name(&1, aliases))
+    |> Enum.uniq()
+  end
+
+  defp filter_namespaces(dependencies, namespaces) do
+    dependencies
```
Comment (rrrene) [inline lib/credo/check/refactor/module_dependencies.ex]: Please remove the pipe on this function call.

File: lib/credo/check/refactor/module_dependencies.ex [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    |> Enum.filter(&keep?(&1, namespaces))
+  end
+
+  defp keep?(_module_name, []), do: true
+
+  defp keep?(module_name, namespaces), do: has_namespace?(module_name, namespaces)
+
+  defp has_namespace?(module_name, namespaces) do
+    namespaces
```
Comment (rrrene) [inline lib/credo/check/refactor/module_dependencies.ex]: Please remove the pipe on this function call.

**Comment by mkorszun** [comment]
@rrrene applied all review fixes
